From bb20c1e1d1726c4d9c7ff313623d7b72d863041a Mon Sep 17 00:00:00 2001
From: "jan.nijtmans" <nijtmans@users.sourceforge.net>
Date: Tue, 22 Jun 2021 09:02:00 +0000
Subject: [PATCH] Fix [bad6cc213d]: A format string vulnerability in Tcl
 nmakehelp.c allows code execution via a crated file. Also change a memcpy()
 to a memmove(), because the range could be overlapping

---
 win/nmakehlp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/win/nmakehlp.c b/win/nmakehlp.c
index 2dc33cc657..f82dc758f7 100644
--- a/win/nmakehlp.c
+++ b/win/nmakehlp.c
@@ -538,8 +538,9 @@ GetVersionFromFile(
 		    ++q;
 		}
 
-		*q = 0;
-		szResult = p;
+		memmove(szBuffer, p, q - p);
+		szBuffer[q-p] = 0;
+		szResult = szBuffer;
 		break;
 	    }
 	}
-- 
2.43.0

